File: Selection Sort/selection_sort.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void selection_sort(int n, int* arr)
{
	int temp;
	for (int i = 0; i < n; i++)
	{
		int smallest_value_index = i;

		for (int j = i; j < n; j++)
		{
			if (arr[j] < arr[smallest_value_index])
			{
				smallest_value_index = j;
			}
		}
		// swap
		temp = arr[i];
		arr[i] = arr[smallest_value_index];
		arr[smallest_value_index] = temp;
	}
}

int main()
{
	int arr_length = 100;
	int arr[100];

	srand(time(0));

	// Add random numbers to array
	for (int i = 0; i < arr_length; i++)
	{
		// Get number in range 0 - 100;
		int num = (rand() % (100 - 0 + 1)) + 0;
		arr[i] = num;
		printf("%d ", arr[i]);
	}

	selection_sort(arr_length, arr);

	printf("\n\nSorted Array: \n");

	// print array
	for (int i = 0; i < arr_length; i++)
	{
		printf("%d ", arr[i]);
	}

	return 0;
}

File: Selection Sort/SelectionSort.java
import java.util.Random;

public class SelectionSort {
    public static void selectionSort(int[] arr) {
        int temp;

        for (int i = 0; i < arr.length; i++) {
            int smallestValueIndex = i;

            for (int j = i; j < arr.length; j++) {
                if (arr[j] < arr[smallestValueIndex]) {
                    smallestValueIndex = j;
                }
            }

            // swap
            temp = arr[i];
            arr[i] = arr[smallestValueIndex];
            arr[smallestValueIndex] = temp;
        }
    }

    public static void main(String[] args) {
        int[] arr = new int[25];

        // Generate random array
        Random rand = new Random();

        // Add random numbers to array
        for (int i = 0; i < arr.length; i++) {
            // Generate random number from 0 to 99;
            int number = rand.nextInt(100);
            arr[i] = number;
            System.out.print(arr[i] + " ");
        }

        selectionSort(arr);
        System.out.println("\n\nSorted Array:");

        // print sorted array
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}
